sql/auth/sql_auth_cache.cc

```diff
@@ -3475,6 +3475,8 @@ void shutdown_acl_cache() {
 
 /* Constants used by Acl_cache_lock_guard */
 static const ulong ACL_CACHE_LOCK_TIMEOUT = 3600UL;
+static const ulong ACL_CACHE_LOCK_TIMEOUT_NSEC =
```

Comment:
`ulong` is not guaranteed to be 64-bit, and therefore may be truncated here as `ACL_CACHE_LOCK_TIMEOUT * 1000000000ULL >2^32`
```suggestion
static constexpr ulonglong ACL_CACHE_LOCK_TIMEOUT_NSEC =
```

Reply:
Thanks, this was one of the typos. I found another when rereading my changes, looks like there aren't any timeout errors so far with these changes.

Updated, and I'll do another update tomorrow with the test changes.